File: 04JavaScript/ex1_3.js
"use strict";
(function () { 

    var calculateLcd = function (dividend, divisor) {
        var remainder = dividend % divisor;
        if (remainder === 0) return divisor;
        return calculateLcd(divisor, remainder);
    }

    var dividend = parseInt(process.argv[2]);
    var divisor = parseInt(process.argv[3]);

    var result = calculateLcd(dividend, divisor);

    console.log(result);
})();

File: 06C/ex1_3.c
#include <stdio.h>

int calculateLcd(int dividend, int divisor)
{
    int remainder = dividend % divisor;

    if (remainder == 0) return divisor;
    calculateLcd(divisor, remainder);
}

int main(int argc, char *argv[])
{
    int dividend = atoi(argv[1]);
    int divisor = atoi(argv[2]);

    int result = calculateLcd(dividend, divisor);

    printf("%i", result);
    return 0;
}